src/soc/intel/common/block/oc_wdt/acpi.c

```diff
@@ -0,0 +1,156 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+
+#include <acpi/acpi.h>
+#include <console/console.h>
+#include <intelblocks/oc_wdt.h>
+
+#define OC_WDT_MIN_VALUE	0
+#define OC_WDT_MAX_VALUE	1023
+#define OC_WDT_PERIOD_MS	1000
+
+static bool fill_wdat_timeout_entry(acpi_wdat_entry_t *entry)
+{
+	memset((void *)entry, 0, sizeof(acpi_wdat_entry_t));
+
+	entry->action = ACPI_WDAT_SET_COUNTDOWN;
+	entry->instruction = ACPI_WDAT_WRITE_COUNTDOWN | ACPI_WDAT_PRESERVE_REGISTER;
+	entry->mask = PCH_OC_WDT_CTL_TOV_MASK;
+	entry->register_region.space_id = ACPI_ADDRESS_SPACE_IO;
+	entry->register_region.addrl = PCH_OC_WDT_CTL;
+	entry->register_region.access_size = ACPI_WDAT_ACCESS_SIZE_DWORD;
```

Comment:
Ste access size to word (16 bits)  as we don't need to write whole register. That way we can also avoid clearing status bits

Reply:
https://github.com/Dasharo/coreboot/pull/472/commits/f99cf95497da3151fe21dc6806c23af9b32e5129